src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/Endpoints/RazorCustomMessageTarget_UpdateCSharpBuffer.cs

```diff
@@ -78,14 +78,25 @@ request.ProjectKeyId is not null &&
             {
                 if (virtualDocument.ProjectKey.Equals(new ProjectKey(request.ProjectKeyId)))
                 {
-                    _logger.LogDebug($"UpdateCSharpBuffer virtual doc for {request.HostDocumentVersion} of {virtualDocument.Uri}");
+                    _logger.LogDebug($"UpdateCSharpBuffer virtual doc for {request.HostDocumentVersion} of {virtualDocument.Uri}. Previous lines: {virtualDocument.Snapshot.LineCount}");
 
                     _documentManager.UpdateVirtualDocument<CSharpVirtualDocument>(
                         hostDocumentUri,
                         virtualDocument.Uri,
                         request.Changes.Select(change => change.ToVisualStudioTextChange()).ToArray(),
                         request.HostDocumentVersion.Value,
                         state: request.PreviousWasEmpty);
+
+                    if (_documentManager.TryGetDocument(hostDocumentUri, out var newDocSnapshot))
+                    {
+                        var newDoc = newDocSnapshot.VirtualDocuments.Where(e => e.Uri == virtualDocument.Uri).FirstOrDefault();
```

Comment:
Calling `Where(...)` followed by `FirstOrDefault()` is a bit of a LINQ anti-pattern. Consider passing the predicate to `FirstOrDefault(...)` instead:

```C#
var newDoc = newDocSnapshot.VirtualDocuments.FirstOrDefault(e => e.Uri == virtualDocument.Uri);
```

Reply:
Thank you for this comment, this code should be removed entirely, it was just added for logging purposes to track down the original issue.

Reply:
Actually, I'll keep it, but move the work to a method so its only called if debug logging is on